### PR TITLE
(DOCSP-43342) Network Security for Atlas Arch Center

### PR DESCRIPTION
**Pull request by lmkerbey-mdb**
[JIRA](https://jira.mongodb.org/browse/DOCSP-43342)
[Staging](https://deploy-preview-30--docs-atlas-architecture.netlify.app/network-security/)

### CODE REVIEW

File: source/network-security.txt [hunk 1]
```diff
@@ -12,18 +12,137 @@ Network Security
    :depth: 1
    :class: onecol
 
-Intro statement
+By default, {+service+} secures your database deployments through the
+use of mandatory TLS/SSL connection encryption, and a VPC-first,
```
Comment (JuliaMongo) [inline source/network-security.txt:16]: we have a replacement for TSL, check shooty.toml please and the compliance topic. We have also a vpc replacement.

File: source/network-security.txt [hunk 2]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+minimum-access default policy. You can further configure these
```
Comment (JuliaMongo) [inline source/network-security.txt:17]: which "default policy" is meant? Did you mean access policy that allows minimum access based on some roles? Did you mean that on top of that access policy we recommend configuring vpc? 
I find this bit very densely packed, we could turn it into a simpler bullet list with links, I think. But, I am not sure this would not change the technical content. So please see what works -- these are just thoughts on how this reads, not a request for change. Hope you can find a middle ground.

File: source/network-security.txt [hunk 3]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+protections to meet your unique security needs and preferences.
 
-{+service+} Features and Best Practices for Network Security
-------------------------------------------------------------
+Use the foundational guidance on this page to plan the network
+security configuration for your {+clusters+}.
 
-Content here
+{+service+} Features and Recommendations for Network Security
```
Comment (JuliaMongo) [inline source/network-security.txt:23]: I believe Sarah wanted to strike Recommendations from these titles now. It is just "features for ABC".

File: source/network-security.txt [hunk 4]
```diff
@@ -12,18 +12,137 @@ Network Security
...
 
-Intro statement
+By default, {+service+} secures your database deployments through the
+use of mandatory TLS/SSL connection encryption, and a VPC-first,
+minimum-access default policy. You can further configure these
+protections to meet your unique security needs and preferences.
 
-{+service+} Features and Best Practices for Network Security
-------------------------------------------------------------
+Use the foundational guidance on this page to plan the network
```
Comment (JuliaMongo) [inline source/network-security.txt:20]: I would remove foundational and instead say: "Use the basic recommendations on this page to plan the network ...."

File: source/network-security.txt [hunk 5]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+security configuration for your {+clusters+}.
 
-Content here
+{+service+} Features and Recommendations for Network Security
+-------------------------------------------------------------
+
+Features
```
Comment (JuliaMongo) [inline source/network-security.txt:26]: Why do we have features again?

File: source/network-security.txt [hunk 6]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+~~~~~~~~
+
+{+service+} enforces TLS/SSL encryption for all connections to your
```
Comment (JuliaMongo) [inline source/network-security.txt:29]: replacement for tls

File: source/network-security.txt [hunk 7]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+databases.
+
+All {+service+} projects with one or more M10+ dedicated {+clusters+}
+receive their own dedicated VPC (or VNet if you use
+|azure|). {+service+} deploys all dedicated clusters inside this VPC
+or VNet. By default, this VPC or VNet allows no inbound access to
+{+service+}, requiring you to explicitly enable access by one of the
+following methods:
```
Comment (JuliaMongo) [inline source/network-security.txt:37]: I would split this sentence in two, and remove the gerund so that our friends the robots can interpret this more easily.

I would also use replacements for VNet and Azure, and to maintain parity with two other cloud providers, I would then add qualifiers for the vpc part (on |aws| and |gcp|).

File: source/network-security.txt [hunk 8]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+- Add public IP addresses to your IP access list
+- Use VPC / VNet peering to add private IP addresses
+- Add private endpoints
+
+TLS
+```
+
+{+service+} enforces mandatory TLS encryption of connections to your
+databases. TLS 1.2 is the default protocol; you can select TLS 1.1 or
+TLS 1.0 if necessary. Note that in situations where TLS 1.1 or later
```
Comment (JuliaMongo) [inline source/network-security.txt:48]: I am totally not sure about earlier versions of TLS. I am pretty sure those are not supported explicitly or were dropped. Please check the docs or search Jira?

File: source/network-security.txt [hunk 9]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+is available, MongoDB 4.0 or later doesn't support TLS 1.0.
```
Comment (JuliaMongo) [inline source/network-security.txt:49]: I would remove the entire bit about MongoDB .40 or later and TLS 1.0. This is something that should be reformulated as follows:

Starting with MongoDB 4.0, |tls| 1.2 is used. Therefore, if you used any other earlier version of |tls|, upgrade your deployments to latest MongoDB versions. Link to migration.

File: source/network-security.txt [hunk 10]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+
+IP Access Lists
+```````````````
+
+You can configure IP Access Lists to limit which IP addresses can
```
Comment (JuliaMongo) [inline source/network-security.txt:54]: do we capitalize IP access list as a feature?

File: source/network-security.txt [hunk 11]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+attempt authentication to your database. Unless you have added the
+relevant IP addresses or CIDR blocks covering the relevant IP
```
Comment (JuliaMongo) [inline source/network-security.txt:56]: cidr has a replacement, use it.

File: source/network-security.txt [hunk 12]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+addresses to your access list, application servers and other clients
+cannot access your {+service+} {+clusters+}.
```
Comment (JuliaMongo) [inline source/network-security.txt:58]: I would rephrase the sentence that starts with "Unless .., to be:

Your {+service+} {+clusters+} allow access only from the relevant IP addresses and IP addresses in |cidr| blocks that you add to the IP access list. Application servers and other clients of MongoDB cannot access your {+service+} {+clusters+} if their IP addresses aren't included in the IP access lists.

File: source/network-security.txt [hunk 13]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+
+You can configure temporary access list entries that expire
+automatically after a user-defined period.
+
```
Comment (JuliaMongo) [inline source/network-security.txt:62]: Here and elsewhere in the previous sections: we need links to docs. I haven't commented but we need them.

File: source/network-security.txt [hunk 14]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+You can create one access list per project.
+
+Firewall Configuration
+``````````````````````
+
+If you use a firewall that blocks outbound network connections, you
+must also configure your firewall to allow your applications to make
+outbound connections to TCP traffic on {+service+} hosts. This grants
+your applications access to databases stored on {+service+}.
```
Comment (JuliaMongo) [inline source/network-security.txt:71]: databases stored on {+service+}.  -- please consider using our standard language for MongoDB Atlas cloud clusters.

File: source/network-security.txt [hunk 15]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+
+{+service+} {+cluster+} public IPs remain the same in the majority of
+cases of {+cluster+} changes: vertical scaling, topology changes,
+maintenance events, healing events, etc. However, certain topology
```
Comment (JuliaMongo) [inline source/network-security.txt:75]: remove etc. What is vertical scaling in this case? What are topology changes? What are maintenance or healing events? Please try to provide simple examples or replace these words with specific scenarios.

File: source/network-security.txt [hunk 16]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+changes – such as a conversion from replica set to sharded cluster, an
```
Comment (JuliaMongo) [inline source/network-security.txt:76]: Links?

File: source/network-security.txt [hunk 17]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+addition of shards, or a region change – require that you use new IP
+addresses.
+
+VPC/VNet Peering
+````````````````
+
+Network peering allows you to connect your own VPCs with an Atlas VPC,
```
Comment (JuliaMongo) [inline source/network-security.txt:83]: replacements

File: source/network-security.txt [hunk 18]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+routing traffic privately and isolating your data flow from the public
+internet. {+service+} does not need outbound access to peer VPCs
```
Comment (JuliaMongo) [inline source/network-security.txt:85]: Internet is capitalized

File: source/network-security.txt [hunk 19]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+except when you have configured LDAPS for authentication. In this
+case, {+service+} {+clusters+} must be able to connect to the LDAPS
+endpoint inside your peer VPC over the LDAPS protocol.
```
Comment (JuliaMongo) [inline source/network-security.txt:88]: Lines 85-88 are very confusing. LDAP, not LDAPS. use a replacement. 
Re: how to make it clearer, I don't have a suggestion, but I think we might have stuff in the docs where this is less confusing.

File: source/network-security.txt [hunk 20]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+
+Recommendations
```
Comment (JuliaMongo) [inline source/network-security.txt:90]: Suggestion: Instead of the breakdown: Features, and then Recommendations, I suggest combining them in the same section, pef feature. As in:
Feature ABC
What's offered by default. 
What can you do in addition.
Recommendations, as a bulleted list with a stem sentence introducing it.

File: source/network-security.txt [hunk 21]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+~~~~~~~~~~~~~~~
+
+IP Access Lists
+```````````````
+
+We recommend using temporary access list entries in situations where
```
Comment (JuliaMongo) [inline source/network-security.txt:96]: Use temporary

File: source/network-security.txt [hunk 22]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+team members require access to your environment from temporary work
+locations.
+
+To reduce the attack surface of your {+service+} projects, we
+recommend defining IP access list entries covering the smallest
```
Comment (JuliaMongo) [inline source/network-security.txt:101]: define

File: source/network-security.txt [hunk 23]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+network segments possible. To do this, favor individual IP
+addresses where possible, and avoid large CIDR blocks.
+
+VPC/VNet Peering
+````````````````
+
+To maintain tight network trust boundaries, configure mitigations such
+as security groups and network ACLs to prevent inbound access to
```
Comment (JuliaMongo) [inline source/network-security.txt:109]: replacements for ACLs, links.

File: source/network-security.txt [hunk 24]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+systems inside your application VPCs from the {+service+}-side VPC.
+
+If your applications rely on VPCs containing extensive unisolated
+infrastructure, you can provide an extra layer of security to these
+systems by deploying new VPCs for those applications that require
+access to {+service+}. These new VPCs can in turn be peered to both
```
Comment (JuliaMongo) [inline source/network-security.txt:115]: remove the if close entirely and shorten.

File: source/network-security.txt [hunk 25]
```diff
@@ -12,18 +12,137 @@ Network Security
...
+
+A private endpoint facilitates a one-way connection from your own VPC
+to your {+service+} VPC, without permitting {+service+} to initiate a
+reciprocal connection. This allows you to make use of secure
+connections to {+service+} without extending your network trust
+boundary. The following private endpoints are available:
+
+- AWS PrivateLink, for connections from AWS VPCs
+- Azure Private Link, for connections from Azure VNets
+- Private Service Connect, for connections from Google Cloud
```
Comment (JuliaMongo) [inline source/network-security.txt:138]: This belongs to the current features section. Add links and replacements.

File: source/network-security.txt [hunk 26]
```diff
@@ -44,10 +163,197 @@ These examples also apply other recommended configurations, including:
...
+	   </connect-atlas-cli/>` using the steps for :guilabel:`Programmatic Use`.
+
+      Create an IP Access List Entry
+      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+      Run the following command for each connection you want to
+      allow. Change the entries to use the appropriate options and
+      your actual values:
+
+      .. code-block::
```
Comment (JuliaMongo) [inline source/network-security.txt:185]: can you add a copy directive?

File: source/network-security.txt [hunk 27]
```diff
@@ -44,10 +163,197 @@ These examples also apply other recommended configurations, including:
...
+
+         atlas accessList create 192.0.2.15 --type ipAddress --projectId 5e2211c17a3e5a48f5497de3 --comment "IP address for app server 2" --output json
+
+      For more configuration options and info about this example, see
```
Comment (JuliaMongo) [inline source/network-security.txt:189]: information

File: source/network-security.txt [hunk 28]
```diff
@@ -44,10 +163,197 @@ These examples also apply other recommended configurations, including:
...
+      :ref:`atlas-accessLists-create`.
+
+      Create a VPC Peering Connection
+      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+      Run the following command for each VPC you want to peer to your
+      {+service+} VPC. Replace ``aws`` with ``azure`` or ``gcp`` as
```
Comment (JuliaMongo) [inline source/network-security.txt:196]: shouldn't we mention VNet?

File: source/network-security.txt [hunk 29]
```diff
@@ -44,10 +163,197 @@ These examples also apply other recommended configurations, including:
...
+      appropriate, and change the options and values to the
+      appropriate ones for your VPC:
+
+      .. code-block::
+
+	 atlas networking peering create aws --accountId 854333054055 --atlasCidrBlock 192.168.0.0/24 --region us-east-1 --routeTableCidrBlock 10.0.0.0/24 --vpcId vpc-078ac381aa90e1e63
+
+      For more configuration options and info about this example, see:
```
Comment (JuliaMongo) [inline source/network-security.txt:204]: information

File: source/network-security.txt [hunk 30]
```diff
@@ -44,10 +163,197 @@ These examples also apply other recommended configurations, including:
...
+      Run the following command for each private endpoint you want to
+      create. Replace ``aws`` with ``azure`` or ``gcp`` as
+      appropriate, and change the options and values to the
+      appropriate ones for your VPC:
+
+      .. code-block::
+
+         atlas privateEndpoints aws create --region us-east-1 --projectId 5e2211c17a3e5a48f5497de3 --output json
+
+      For more configuration options and info about this example, see:
```
Comment (JuliaMongo) [inline source/network-security.txt:222]: information

**Review by JuliaMongo — COMMENTED** [review]
I will review again. Thank you, Lander! Tons of work here.

File: source/network-security.txt (hunk 8 above)
Comment (lmkerbey-mdb) [reply source/network-security.txt:48]: Added a reference to a section of the Atlas docs that talks about this issue already.

File: source/network-security.txt (hunk 10 above)
Comment (lmkerbey-mdb) [reply source/network-security.txt:54]: Created a source constant to enforce consistent capitalization based on grep results in cloud-docs (233 occurrences of IP access list, 24 of IP Access List), replaced as needed.

**Comment by JuliaMongo** [comment]
Back to you with just a few small comments. The main issue is that many links don't work or are malformed - show paths in the staging link page. Please fix them and I will re-check again to make sure all of them work. Lots and lots of work! Thank you, Lander!
